Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -282,8 +282,8 @@ private static bool HasValidAccessModifier(PropertyInfo propertyInfo)
         {
             var methodInfo = propertyInfo.GetGetMethod(true);
 
-            return methodInfo != null && ((methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly))
-                || propertyInfo.GetCustomAttribute(typeof(IonPropertyName)) != null;
+            return methodInfo != null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly
```

Comment:
Is this correct? This will exclude the method if there is no `methodInfo` even if it's annotated which seems incorrect, although TBH I don't know when `methodInfo` will be null.

Reply:
A property can potentially contain two methods (getter & setter). This function checks if `methodInfo` aka getter is defined.

Now currently this function is used to filter properties during serialization & deserialization.

For serialization if there is no getter, then we cannot retrieve the value so we will we want to filter out the property regardless of annotation.

For deserialization, if there is no setter, then we cannot set the value to the property.

I will make the change to create a separate method to ensure a property has a setter before we attempt to set the value.

Reply:
Updated in this [commit](https://github.com/amzn/ion-object-mapper-dotnet/pull/36/commits/d7d4170b6b2b0dddafbdd74618687eee9052fd74)